plugin/builtin/replicacounttransformer/ReplicaCountTransformer_test.go

```diff
@@ -148,9 +149,8 @@ spec:
 }
 
 func TestMatchesCurrentID(t *testing.T) {
-	th := kusttest_test.MakeEnhancedHarness(t).
-		PrepBuiltin("PrefixSuffixTransformer").
```

Comment:
I see that in this test we are testing the legacy `PrefixSuffixTransformer` behavior, but I think it would be good to have a separate explicit test. I was wondering if you could add a test to `api/krusty`, perhaps in a new file called `legacyprefixsuffixtransformer_test.go`, to test the following kustomization file:

```
resources:
- service.yaml
transformers: 
- |- 
  apiVersion: builtin
  kind: PrefixSuffixTransformer
  metadata:
    name: notImportantHere
  prefix: baked-
  suffix: -pie
  fieldSpecs:
    - path: metadata/name
```

You can look at some other test files in `api/krusty` as examples, [TestBasicIO_1](https://github.com/kubernetes-sigs/kustomize/blob/ec38bbeb999066abea59fd11c54e68377765a510/api/krusty/basic_io_test.go#L12) is probably one where you could copy/paste the entire test and just change the input and expected output files to write this test. As described in [my original comment](https://github.com/kubernetes-sigs/kustomize/issues/4164#issuecomment-981920422) on the issue, we need to make sure we still support this use case for a while.

Reply:
I added the test in `api/krusty/legacyprefixsuffixtransformer_test.go` 👍🏼